app/auth/authV2/actions/AsMtdUser.scala

```diff
@@ -72,26 +91,27 @@ class AsMtdUser @Inject()
 
     implicit val r = request
 
-    val (optMtdId, optClientName) = request.affinityGroup match {
-      case Some(Agent) => (getClientMtdid, getClientName)
-      case _ => (Future.successful(request.mtdId), None)
+    for {
+      mtdAndUtr <- getClientMtdidAndUtr
+      clientName <- getClientName(mtdAndUtr._2)
```

Comment:
if we can avoid using tuple in this way: ._2 / _1 as it hard to read this code later;

Reply:
I'm trying to make it so that we only have to call the SessionDataService once, but yeah I'll try to make this more readable